extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubePlaylistExtractor.java

```diff
@@ -1,85 +1,117 @@
 
+import com.grack.nanojson.JsonArray;
+import com.grack.nanojson.JsonObject;
+import com.grack.nanojson.JsonParser;
+import com.grack.nanojson.JsonParserException;
 import org.schabi.newpipe.extractor.StreamingService;
 import org.schabi.newpipe.extractor.downloader.Downloader;
+import org.schabi.newpipe.extractor.downloader.Response;
 import org.schabi.newpipe.extractor.exceptions.ExtractionException;
 import org.schabi.newpipe.extractor.exceptions.ParsingException;
 import org.schabi.newpipe.extractor.linkhandler.ListLinkHandler;
 import org.schabi.newpipe.extractor.playlist.PlaylistExtractor;
+import org.schabi.newpipe.extractor.services.peertube.PeertubeParsingHelper;
 import org.schabi.newpipe.extractor.stream.StreamInfoItem;
+import org.schabi.newpipe.extractor.stream.StreamInfoItemsCollector;
+import org.schabi.newpipe.extractor.utils.JsonUtils;
 
+import javax.annotation.Nonnull;
 import java.io.IOException;
 
+import static org.schabi.newpipe.extractor.services.peertube.PeertubeParsingHelper.*;
+
 public class PeertubePlaylistExtractor extends PlaylistExtractor {
 
+    private JsonObject playlistInfo;
+    private JsonObject playlistVideos;
+
+    private InfoItemsPage<StreamInfoItem> initPage;
+    private long total;
+
     public PeertubePlaylistExtractor(StreamingService service, ListLinkHandler linkHandler) {
         super(service, linkHandler);
-        // TODO Auto-generated constructor stub
     }
 
     @Override
     public String getThumbnailUrl() throws ParsingException {
-        // TODO Auto-generated method stub
-        return null;
+        return getBaseUrl() + playlistInfo.getString("thumbnailPath");
     }
 
     @Override
     public String getBannerUrl() throws ParsingException {
-        // TODO Auto-generated method stub
         return null;
     }
 
     @Override
     public String getUploaderUrl() throws ParsingException {
-        // TODO Auto-generated method stub
-        return null;
+        return playlistInfo.getObject("ownerAccount").getString("url");
     }
 
     @Override
     public String getUploaderName() throws ParsingException {
-        // TODO Auto-generated method stub
-        return null;
+        return playlistInfo.getObject("ownerAccount").getString("displayName");
     }
 
     @Override
     public String getUploaderAvatarUrl() throws ParsingException {
-        // TODO Auto-generated method stub
-        return null;
+        return getBaseUrl() + playlistInfo.getObject("ownerAccount").getObject("avatar").getString("path");
     }
 
     @Override
     public long getStreamCount() throws ParsingException {
-        // TODO Auto-generated method stub
-        return 0;
+        return playlistInfo.getNumber("videosLength").longValue();
     }
 
+    @Nonnull
     @Override
     public InfoItemsPage<StreamInfoItem> getInitialPage() throws IOException, ExtractionException {
-        // TODO Auto-generated method stub
-        return null;
+        super.fetchPage();
```

Comment:
When `getInitialPage()` is called, `fetchPage()` should always already have been called, so no need to call it here. You also shouldn't fetch the initial page in `onFetchPage()` yet, but only in `getInitialPage()` and `getNextPageUrl()` if it hasn't been fetched yet.

I see that something similar is done in some other services though, but that should also be changed, and imho we should just remove `getNextPageUrl()`.